pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -426,7 +539,6 @@ def _set_objective(self, obj):
             self._objective = None
 
         self._solver_model.objective.set_linear([(i, 0.0) for i in range(len(self._pyomo_var_to_solver_var_map.values()))])
-        self._solver_model.objective.set_quadratic([[[0], [0]] for i in self._pyomo_var_to_solver_var_map.keys()])
```

Comment:
@ruaridhw I don't think this line should be removed completely. However, it can be simplified to:

```
self._solver_model.objective.set_quadratic([0]*len(self._pyomo_var_to_solver_var_map))
```

If this line is not included, then the quadratic part of the objective will not be updated correctly between solves (for the persistent interface).

Reply:
@michaelbynum, could you provide a small example of how this would break? I'd like to add a test case to that effect.

Reply:
@ruaridhw  This should demonstrate the issue:

```
import pyomo.environ as pe

m = pe.ConcreteModel()
m.x = pe.Var(bounds=(-2, 2))
m.y = pe.Var(bounds=(-2, 2))
m.obj = pe.Objective(expr=m.x**2 + m.y**2)
m.c1 = pe.Constraint(expr=m.y >= 2*m.x - 1)
m.c2 = pe.Constraint(expr=m.y >= -m.x + 2)
opt = pe.SolverFactory('cplex_persistent')
opt.set_instance(m)
opt.solve()
print(m.x.value, m.y.value)  # should be 1, 1
del m.obj
m.obj = pe.Objective(expr=m.x**2)
opt.set_objective(m.obj)
opt.solve()
print(m.x.value, m.y.value)  # should be 0, 2 but result is 1, 1
opt.set_instance(m)
opt.solve()
print(m.x.value, m.y.value)  # to demonstrate that the result should be 0, 2
```

Reply:
@michaelbynum, I've added a fix and this test case. The main issue I had was that the quadratic objective interface shouldn't be triggered if the model is a MILP. Let me know if 1b555dc is acceptable.